control-plane/agents/core/src/core/scheduling/mod.rs

```diff
@@ -72,6 +75,42 @@ impl PoolFilters {
     pub(crate) fn usable(_: &GetSuitablePoolsContext, item: &PoolItem) -> bool {
         item.pool.status != PoolStatus::Faulted && item.pool.status != PoolStatus::Unknown
     }
+    /// Should only attempt to use pools having specific creation label iff topology has it
+    pub(crate) fn topology(request: &GetSuitablePoolsContext, item: &PoolItem) -> bool {
+        let volume_creation_label: String;
+        match request.topology.clone() {
+            None => return true,
+            Some(topology) => match topology.pool {
+                None => return true,
+                Some(pool_topology) => match pool_topology {
+                    PoolTopology::Labelled(labelled_topology) => {
+                        if labelled_topology
+                            .inclusion
+                            .contains_key(OPENEBS_CREATED_BY_KEY)
+                        {
+                            volume_creation_label = labelled_topology
+                                .inclusion
+                                .get(OPENEBS_CREATED_BY_KEY)
+                                .unwrap()
+                                .clone();
+                        } else {
+                            return true;
```

Comment:
It might be a good idea to include a comment here stating that we don't support other keys for now. In other words, if any other label is set on the pool, we will ignore it and use any pool.

Reply:
Yep. Added the comment for the same.